### PR TITLE
[Build] Add seatunnel-spark-3-starter.jar into release package

### PR DESCRIPTION
**Pull request by Hisoka-X**
### Purpose of this pull request
This PR fix release package not included seatunnel-spark-3-starter.jar because it scope isnot provided.

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
tested in local

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-dist/pom.xml [hunk 1]
```diff
@@ -720,6 +720,7 @@
                     <groupId>org.apache.seatunnel</groupId>
                     <artifactId>seatunnel-spark-3-starter</artifactId>
                     <version>${project.version}</version>
+                    <scope>provided</scope>
```
Comment (EricJoy2048) [inline seatunnel-dist/pom.xml:717]: I am not found the spark and flink License in `seatunnel-dist/release-docs/LICENSE`. It is necessary if you put `seatunnel-spark-3-starter` in release package.

Comment (Hisoka-X) [reply seatunnel-dist/pom.xml:717]: done

**Review by hailin0 — COMMENTED** [review]
LGTM
waiting for ci passed

**Review by hailin0 — APPROVED** [review]

**Review by ruanwenjun — APPROVED** [review]
+1

**Pull request merged**